roll/__init__.py

```diff
@@ -199,10 +199,12 @@ def write(self, *args):
         payload = b'HTTP/1.1 %a %b\r\n' % (
             self.response.status.value, self.response.status.phrase.encode())
         if not isinstance(self.response.body, bytes):
-            self.response.body = self.response.body.encode()
+            self.response.body = str(self.response.body).encode()
         if 'Content-Length' not in self.response.headers:
             length = len(self.response.body)
             self.response.headers['Content-Length'] = length
+        if 'Content-Type' not in self.response.headers:
```

Comment:
I'm not sure this really serve a use case. Plus it costs a `not in` for each request (not a big deal but…).
I'd better let the user explicitly set it.
Or at least have a `DEFAULT_CONTENT_TYPE` property, so it can be changed.

Reply:
I set a default.

Reply:
After a second thought, I really think this can be treacherous. IMHO Content-Type should be set explicitly (or using dedicated shortcuts like `json`).

Reply:
> Any HTTP/1.1 message containing an entity-body SHOULD include a
Content-Type header field defining the media type of that body. If
and only if the media type is not given by a Content-Type field, the
recipient MAY attempt to guess the media type via inspection of its
content and/or the name extension(s) of the URI used to identify the
resource. If the media type remains unknown, the recipient SHOULD
treat it as type "application/octet-stream".

I really think as a web framework we should set a default, if it's easy to override both per view and for the whole app.

Reply:
> SHOULD

Means it's valid without ;)

But my point is that we are implicitly setting a value, and we have only one chance over X that this value is useful. So I think setting the correct value, per view, is the role of the user, not the one of the framework (which is blind on this topic).
*If* you really want to be educational, raise if the header is not found (but that would not be correct given the header is not mandatory as per the HTTP spec you quoted above) ;)